crates/sargon/src/signing/signatures_outecome_types/petition_transaction_outcome.rs

```diff
@@ -5,29 +5,29 @@ use crate::prelude::*;
 /// set of collected signatues (might be empty) and
 /// a set of neglected factors (might be empty).
 #[derive(Clone, PartialEq, Eq)]
-pub(crate) struct PetitionTransactionOutcome {
-    intent_hash: TransactionIntentHash,
+pub(crate) struct PetitionTransactionOutcome<S: Signable> {
```

Comment:
Looks like this can be made generic over `T: SignableID` only, not the whole `Signable` trait?